orchard-ws/app/controllers/WorkflowController.scala

```diff
@@ -73,6 +73,30 @@ class WorkflowController @Inject() (
     )
   }
 
+  private def validateResources(request: WorkflowRequest): Either[String, WorkflowRequest] = {
+
+    val invalidTerminateAt =
+      for {
+        resource <- request.resources
+        terminateAt <- resource.terminateAfter
+        if terminateAt <= 0 || terminateAt > 24 * 30
+      } yield terminateAt
+
+    val invalidMaxAttempts =
+      for {
+        resource <- request.resources
+        if resource.maxAttempt <= 0 || resource.maxAttempt > 100
+      } yield resource.maxAttempt
+
+    if (invalidTerminateAt.nonEmpty) {
+      Left("Invalid terminateAt value")
+    } else if (invalidMaxAttempts.nonEmpty) {
+      Left("Invalid maxAttempt value")
+    } else {
+      Right(request)
+    }
+  }
```

Comment:
@realstraw don't know what is a better way of doing this. Feel free to comment for improvements.

Reply:
can probably just call this in `processWorkflowRequest` instead and make it return `Future[Either]`.

Also, it'd be good to make the error message containing more detail. e.g. "Invalid terminateAt value, must be a number between 0 and 24 * 30"`

Reply:
you mean make `processWorkflowRequest` return `Either[String, Future]` right?

Reply:
hmm either way is kind of awkward.

Reply:
I think the previous version is cleaner

Reply:
ok, let's use the previous version then

Reply:
okay, reverted, added more detailed error messages

Reply:
thanks!